BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -106,186 +111,190 @@ public class TransactionConfirmationStore: ObservableObject {
-            self.fetchAssetRatios(for: self.state.symbol, gasSymbol: self.state.gasSymbol)
+      case let .ethErc20Approve(details):
+        state.value = details.approvalAmount
+        state.symbol = details.token.symbol
+        state.proposedAllowance = details.approvalValue
+        state.isUnlimitedApprovalRequested = details.isUnlimited
+        if let gasFee = details.gasFee {
+          state.gasValue = gasFee.fee
+          state.gasFiat = gasFee.fiat
+          state.gasSymbol = parsedTransaction.networkSymbol
+          state.gasAssetRatio = assetRatios[parsedTransaction.networkSymbol.lowercased(), default: 0]
+          
+          if let gasBalance = gasTokenBalance,
+             let gasValue = BDouble(gasFee.fee),
+             BDouble(gasBalance) > gasValue {
+            state.isBalanceSufficient = true
+          } else {
+            state.isBalanceSufficient = false
           }
         }
-      }
-    }
-  }
-
-  private func fetchAssetRatios(for symbol: String, gasSymbol: String) {
-    @discardableResult func updateState() -> Bool {
-      if let ratio = assetRatios[symbolKey], let gasRatio = assetRatios[gasKey] {
-        let value = (Double(self.state.value) ?? 0.0) * ratio
-        self.state.fiat = currencyFormatter.string(from: NSNumber(value: value)) ?? ""
-        let gasValue = (Double(self.state.gasValue) ?? 0.0) * gasRatio
-        self.state.gasFiat = currencyFormatter.string(from: NSNumber(value: gasValue)) ?? ""
-        self.state.totalFiat = currencyFormatter.string(from: NSNumber(value: value + gasValue)) ?? ""
-        self.state.gasAssetRatio = gasRatio
-        return true
-      }
-      return false
-    }
-    let symbolKey = symbol.lowercased()
-    let gasKey = gasSymbol.lowercased()
-    if updateState() {
-      isLoading = false
-    } else {
-      let symbols = symbolKey == gasKey ? [symbolKey] : [symbolKey, gasKey]
-      assetRatioService.price(
-        symbols,
-        toAssets: [currencyFormatter.currencyCode],
-        timeframe: .oneDay
-      ) { [weak self] success, prices in
-        // `success` only refers to finding _all_ prices and if even 1 of N prices
-        // fail to fetch it will be false
-        guard let self = self,
-          self.state.symbol.caseInsensitiveCompare(symbol) == .orderedSame
-        else {
-          return
-        }
-        if let price = prices.first(where: { $0.fromAsset == symbolKey }),
-          let ratio = Double(price.price) {
-          self.assetRatios[symbolKey] = ratio
+        
+        state.totalFiat = state.gasFiat
+        
+        // Update `State.currentAllowance`
+        let formatter = WeiFormatter(decimalFormatStyle: .balance)
+        let contractAddress = transaction.txDataUnion.ethTxData1559?.baseData.to ?? ""
+        if let token = allTokens.first(where: {
+          $0.contractAddress(in: selectedChain).caseInsensitiveCompare(contractAddress) == .orderedSame
+        }) {
+          rpcService.erc20TokenAllowance(
```

Comment:
I think you should be able to use the async/await version of this function here. You likely want to add `spenderAddress: String` to `EthErc20ApproveDetails` and include it in the `TransactionParser` for `erc20Approve` type. Then you should be able to use async/await like so:

```
        let (allowance, _, _) = await rpcService.erc20TokenAllowance(
          details.token.contractAddress(in: network),
          ownerAddress: parsedTransaction.fromAddress,
          spenderAddress: details.spenderAddress
        )
        let formatter = WeiFormatter(decimalFormatStyle: .balance)
        self.state.currentAllowance = formatter.decimalString(for: allowance.removingHexPrefix, radix: .hex, decimals: Int(details.token.decimals)) ?? ""
```

We could also optionally pass in the current allowance similar to the `solEstimatedTxFee: UInt64?`, but I'm not sure necessary as it's only used in this one view / store.